library/core/src/time.rs

```diff
@@ -258,6 +277,8 @@ impl Duration {
     #[must_use]
     #[inline]
     #[rustc_const_stable(feature = "duration_consts", since = "1.32.0")]
+    // by definition of MILLIS_PER_SEC, the checks for div by 0 cases are unreachable, but also unneeded
```

Comment:
As above: should you perhaps just spell this out as a precondition, even though it is trivially satisfied (at the moment!)?